llvm/lib/Target/SystemZ/SystemZInstrInfo.cpp

```diff
@@ -1883,6 +1931,10 @@ void SystemZInstrInfo::getLoadStoreOpcodes(const TargetRegisterClass *RC,
   } else if (RC == &SystemZ::FP128BitRegClass) {
     LoadOpcode = SystemZ::LX;
     StoreOpcode = SystemZ::STX;
+  } else if (RC == &SystemZ::FP16BitRegClass ||
+             RC == &SystemZ::VR16BitRegClass) {
+    LoadOpcode = SystemZ::VL16;
```

Comment:
Yes, for instance in spill-half-01.mir.

Reply:
What I meant is that `VL16` is defined as having a `v16hb` return value
```
    def VL16 : UnaryAliasVRX<z_load, v16hb, bdxaddr12pair>;
```
and `v16hb` is defined to live in `VR16` - *not* `FP16`:
```
def v16hb   : TypedReg<f16,     VR16>;
```

So it seems to me forcing a `FP16` register into the instruction, even if it might seem to do what we want, is not really allowed and might e.g. trigger strict MI checking failures ...

Reply:
I was assuming this would work, as FP16 should be a sub-registerclass of VR16 - all FP16 regs are also contained in VR16. I see in the generated file:

```
static const TargetRegisterClass *const FP16BitSuperclasses[] = {
  &SystemZ::VR16BitRegClass,
  nullptr
};
```
Given that, and that -verify-machineinstrs is used in spill-half-01.mir, it seems ok to me.

Reply:
Ah OK.  That's probably fine then.